tests/test_autodiff.py

```diff
@@ -46,6 +46,59 @@ def operator(kspace_loc, shape, backend, autograd):
     return nufft
 
 
+def proper_trajectory_torch(trajectory, normalize="pi"):
+    """Normalize the trajectory to be used by NUFFT operators on device."""
+    if not torch.is_tensor(trajectory):
+        raise ValueError("trajectory should be a torch.Tensor")
+
+    new_traj = trajectory.clone()
+    new_traj = new_traj.view(-1, trajectory.shape[-1])
+
+    if normalize == "pi" and torch.max(torch.abs(new_traj)) - 1e-4 < 0.5:
+        warnings.warn(
+            "Samples will be rescaled to [-pi, pi), assuming they were in [-0.5, 0.5)"
+        )
+        new_traj *= 2 * torch.pi
+    elif normalize == "unit" and torch.max(torch.abs(new_traj)) - 1e-4 > 0.5:
+        warnings.warn(
+            "Samples will be rescaled to [-0.5, 0.5), assuming they were in [-pi, pi)"
+        )
+        new_traj /= 2 * torch.pi
+
+    if normalize == "unit" and torch.max(new_traj) >= 0.5:
+        new_traj = (new_traj + 0.5) % 1 - 0.5
+
+    return new_traj
+
+
+def get_fourier_matrix_torch(ktraj, shape, dtype=torch.complex64, normalize=False):
```

Comment:
This too.. I think we need a better way to handle this in tests and this could potentially open new issues. We need the trajectory to be a tensor, can mri-nufft handle it? Even if it is on GPU?